packages/webapp/pages/account/security.tsx

```diff
@@ -2,16 +2,82 @@ import TabContainer, {
   Tab,
 } from '@dailydotdev/shared/src/components/tabs/TabContainer';
 import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
-import React, { ReactElement, useContext, useState } from 'react';
+import { formToJson } from '@dailydotdev/shared/src/lib/form';
+import usePrivilegedSession from '@dailydotdev/shared/src/hooks/usePrivilegedSession';
+import VerifySessionModal from '@dailydotdev/shared/src/components/auth/VerifySessionModal';
+import { useToastNotification } from '@dailydotdev/shared/src/hooks/useToastNotification';
+import React, {
+  FormEvent,
+  ReactElement,
+  useContext,
+  useRef,
+  useState,
+} from 'react';
+import { useMutation, useQuery } from 'react-query';
+import {
+  AuthFlow,
+  initializeKratosFlow,
+  submitKratosFlow,
+} from '@dailydotdev/shared/src/lib/kratos';
+import {
+  getNodeValue,
+  ValidateResetPassword,
+} from '@dailydotdev/shared/src/lib/auth';
 import { AccountSecurityDisplay as Display } from '../../components/layouts/AccountLayout/common';
 import { getAccountLayout } from '../../components/layouts/AccountLayout';
 import AccountSecurityDefault from '../../components/layouts/AccountLayout/Security';
 import EmailFormPage from '../../components/layouts/AccountLayout/Security/EmailFormPage';
+import ChangePasswordForm, {
+  ChangePasswordParams,
+} from '../../components/layouts/AccountLayout/Security/ChangePasswordForm';
+
+interface ResetFormParams {
+  password: string;
+}
 
 const AccountSecurityPage = (): ReactElement => {
   const { user } = useContext(AuthContext);
   const [isEmailSent, setIsEmailSent] = useState(false);
   const [activeDisplay, setActiveDisplay] = useState(Display.Default);
+  const { displayToast } = useToastNotification();
+  const formRef = useRef<HTMLFormElement>();
+  const onPasswordReset = () => {
+    displayToast('Password reset successful!');
+    setActiveDisplay(Display.Default);
+  };
+  const { data: settings } = useQuery(['settings'], () =>
+    initializeKratosFlow(AuthFlow.Settings),
+  );
+
+  const {
+    showVerifySession,
+    initializePrivilegedSession,
+    onPasswordLogin,
+    onCloseVerifySession,
+    onSocialLogin,
+  } = usePrivilegedSession();
+
+  const { mutateAsync: resetPassword } = useMutation(
+    (params: ValidateResetPassword) => submitKratosFlow(params),
+    {
+      onSuccess: async ({ redirect }) => {
+        if (redirect) {
+          return initializePrivilegedSession(redirect);
+        }
```

Comment:
I think we have to distinguish if it's a privileged or not.
You could still be valid not to have privileged,  but still need to redirect.

Reply:
Ah, good point. We need to check the given url if it is a login one.

Reply:
We can still call this now, but I have placed a condition checker in the custom hook so we won't have to check it every implementation.